st2reactor/st2reactor/container/sensor_wrapper.py

```diff
@@ -67,6 +73,106 @@ def dispatch(self, trigger, payload=None):
         """
         self._dispatcher.dispatch(trigger, payload=payload)
 
+    def get_value(self, name):
+        """
+        Retrieve a value from the datastore for the provided key.
+
+        :param name: Key name.
+        :type name: ``str``
+
+        :rtype: ``str`` or ``None``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        try:
+            kvp = client.keys.get_by_id(id=name)
+        except Exception:
+            return None
+
+        if kvp:
+            return kvp.value
+
+        return None
+
+    def set_value(self, name, value):
+        """
+        Set a value for the provided key.
+
+        :param name: Key name.
+        :type name: ``str``
+
+        :param value: Key value.
+        :type value: ``str``
+
+        :return: ``True`` on sucess, ``False`` otherwise.
+        :rtype: ``bool``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        instance = KeyValuePair()
+        instance.id = name
+        instance.name = name
+        instance.value = value
+        client.keys.update(instance=instance)
+        return True
+
+    def delete_value(self, name):
+        """
+        Delete the provided key.
+
+        :param name: Name of the key to delete.
+        :type name: ``str``
+
+        :return: ``True`` on sucess, ``False`` otherwise.
+        :rtype: ``bool``
+        """
+        name = self._get_full_key_name(name=name)
+        client = self._get_api_client()
+
+        instance = KeyValuePair()
+        instance.id = name
+        instance.name = name
+
+        try:
+            client.keys.delete(instance=instance)
+        except Exception:
+            return False
+
+        return True
+
+    def _get_api_client(self):
+        """
+        Retrieve API client instance.
+        """
+        # TODO: API client is really unfriendly and needs to be re-designed and
+        # improved
+        api_url = os.environ.get(API_URL_ENV_VARIABLE_NAME, None)
+        auth_token = os.environ.get(AUTH_TOKEN_ENV_VARIABLE_NAME, None)
+
+        if not api_url or not auth_token:
+            raise ValueError('%s and %s environment variable must be set' %
+                             (API_URL_ENV_VARIABLE_NAME, AUTH_TOKEN_ENV_VARIABLE_NAME))
+
+        if not self._client:
+            self._client = Client(api_url=api_url)
```

Comment:
Yeah this bothered me a lot as well. Also, at places the auth token is assigned via decorators and that does not make a great deal of sense while consuming the client. In the CLI though the decorator approach works quite well.